app/controllers/lti_deployments_controller.rb

```diff
@@ -180,11 +180,18 @@ def check_host
   end
 
   def create_course
-    new_course = Course.create!(name: params['name'], display_name: params['display_name'], is_hidden: true)
+    new_course = Course.find_or_initialize_by(name: params['name'], display_name: params['display_name'])
```

Comment:
It is only the course `name` that is unique. You should first do a search just on `name`, and then (if a new record) update both `display_name` and `is_hidden`